examples/Conversational_agents_using_tools.ipynb

```diff
@@ -0,0 +1,1095 @@
+    "\n",
+    "# LLM chain consisting of the LLM and a prompt\n",
+    "llm_chain = LLMChain(llm=llm, prompt=prompt)\n",
+    "\n",
+    "# Using tools, the LLM chain and output_parser to make an agent\n",
+    "tool_names = [tool.name for tool in tools]\n",
+    "\n",
+    "agent = LLMSingleActionAgent(\n",
+    "    llm_chain=llm_chain, \n",
+    "    output_parser=output_parser,\n",
+    "    # We use \"Observation\" as our stop sequence so it will stop when it receives Tool output\n",
+    "    # If you change your prompt template you'll need to adjust this as well\n",
+    "    stop=[\"\\nObservation:\"], \n",
+    "    allowed_tools=tool_names\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "dd893fa5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initiate the agent that will respond to our queries\n",
+    "# Set verbose=True to share the CoT reasoning the LLM goes through\n",
+    "agent_executor = AgentExecutor.from_agent_and_tools(agent=agent, tools=tools, verbose=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "c7ca3ee2",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\n",
+      "\u001B[1m> Entering new AgentExecutor chain...\u001B[0m\n",
+      "\u001B[32;1m\u001B[1;3mThought: Hmm, I be not sure of the answer to that one. Let me think.\n",
+      "Action: Search\n",
+      "Action Input: \"Canada population 2023\"\u001B[0m\n",
+      "\n",
+      "Observation:\u001B[36;1m\u001B[1;3mThe current population of Canada is 38,662,830 as of Monday, April 17, 2023, based on Worldometer elaboration of the latest United Nations data.\u001B[0m\u001B[32;1m\u001B[1;3mAhoy, that be the answer I was lookin' for!\n",
+      "Final Answer: The population of Canada as of 2023 be 38,662,830, Arg!\u001B[0m\n",
+      "\n",
+      "\u001B[1m> Finished chain.\u001B[0m\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "'The population of Canada as of 2023 be 38,662,830, Arg!'"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "agent_executor.run(\"How many people live in canada as of 2023?\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c3b51817",
+   "metadata": {},
+   "source": [
+    "## LLM Agent with History\n",
+    "\n",
+    "Extend the LLM Agent with the ability to retain a [memory](https://python.langchain.com/en/latest/modules/agents/agents/custom_llm_agent.html#adding-memory) and use it as context as it continues the conversation.\n",
+    "\n",
+    "We use a simple ConversationBufferWindowMemory for this example that keeps a rolling window of the last two conversation turns, but LangChain have many other [memory options](https://python.langchain.com/en/latest/modules/memory.html) depending on your use case."
```

Comment:
nit: Technically, the existence of LangChain's memory options doesn't depend on use case (as they exist either way). Really the thing that depends on the use case is which option you choose. Could rephrase sentence to something like "LangChain has many other memory options, with different tradeoffs suitable for different use cases."